lib/ledger_sync/adaptors/quickbooks_online/journal_entry/operations/create.rb

```diff
@@ -0,0 +1,56 @@
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module JournalEntry
+        module Operations
+          class Create < Operation::Create
+            class Contract < LedgerSync::Adaptors::Contract
+              params do
+                required(:ledger_id).value(:nil)
+                optional(:currency).filled(:string)
+                optional(:memo).filled(:string)
+                optional(:transaction_date).filled(:date?)
+                optional(:line_items).array(Types::Reference)
+              end
+            end
+
+            private
+
+            def operate
+              response = adaptor.post(
+                resource: 'journal_entry',
+                payload: local_resource_data
+              )
+
+              resource.ledger_id = response.dig('Id')
+              success(response: response)
+            end
+
+            def local_resource_data
+              {
+                'CurrencyRef' => {
+                  'value' => resource.currency,
+                },
+                'TxnDate' => resource.transaction_date.to_s, # Format: YYYY-MM-DD
+                'PrivateNote' => resource.memo,
+                'Line' => resource.line_items.map do |line_item|
+                  {
+                    'DetailType' => 'JournalEntryLineDetail',
+                    'JournalEntryLineDetail' => {
+                      'PostingType': Mapping::LINE_ITEM_TYPES[line_item.entry_type],
+                      'AccountRef' => {
+                        'value' => line_item.account&.ledger_id
```

Comment:
Why make this nil safe?

Reply:
In case we don't have account, right now we will let QBO fail, instead of crashing in the lib. This will change once we handle serialisation and dirty.

Right now we would send `nil` to QBO, which could remove account if its optional attribute. With serialisation we will not serialise it if we don't provide account.